todo_app/to_do_playbook.yml

```
@@ -0,0 +1,71 @@
+- name: To-do app config
+  hosts: webservers
+  remote_user: ec2-user
+
+  tasks:
+  - name: install git
+    yum:
+      name: git
+      state: present
+    become: yes  
+
+  - name: install python 3
+    yum:
+      name: python3
+      state: present
+    become: yes  
+
+  - name: install poetry if not already installed
+    ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+    args:
+      creates: ~/.local/bin/poetry
+
+- name: Set up and deploy to-do app
+  hosts: webservers
+  remote_user: ec2-user
+  vars_prompt:
+    - name: secret_key
+      prompt: what is the Flask secret key?
+      private: yes 
+    - name: trello_key
+      prompt: what is the trello key?
+      private: yes 
+    - name: trello_token
+      prompt: what is the trello token?
+      private: yes 
+    - name: trello_board_id
+      prompt: what is the board id?
+      private: yes 
+    - name: trello_open_list_id
+      prompt: what is the open list id?
+      private: yes 
+    - name: trello_closed_list_id
+      prompt: what is the closed list id?
+      private: yes 
+
+  tasks:
+  - name: create directory /opt/todoapp with owner permissions
+    ansible.builtin.file:
+      path: /opt/todoapp
+      state: directory
+      mode: '755'
+      owner: ec2-user
+    become: yes  
+
+  - name: checkout latest version of code from git
+    ansible.builtin.git:
+      repo: 'https://github.com/Philw10/DevOps-Course-Starter.git'
+      dest: /opt/todoapp
+      version: module4 
+
+  - name: install project dependencies
+    ansible.builtin.shell: cd /opt/todoapp; ~/.local/bin/poetry install
+
+  - name: create .env
+    ansible.builtin.template:
+      src: /home/ec2-user/.env.j2
+      dest: /opt/todoapp/.env
+      owner: ec2-user
+      mode: '700'
+
```

I think you're missing your last couple of plays, but logging onto the control node I can see you have set them up, so it's just a matter of making sure they're tracked in Git

```suggestion

  - name: copy service file to system
    ansible.builtin.copy:
      src: /opt/todoapp/todoapp.service
      dest: /etc/systemd/system/todoapp.service
      remote_src: yes
    become: yes

  - name: start app using systemd
    ansible.builtin.systemd:
      state: started
      name: todoapp
      daemon_reload: true
    become: yes
```